rascal-lsp/src/main/rascal/library/util/LanguageServer.rsc

```diff
@@ -270,8 +270,17 @@ data LanguageService
     | implementation(set[loc] (Focus _focus) implementationService)
     | codeAction    (list[CodeAction] (Focus _focus) codeActionService)
     | selectionRange(list[loc](Focus _focus) selectionRangeService)
+    | formatting    (list[TextEdit](Tree _input, loc range, FormattingOptions _opts) formattingService)
     ;
 
+data FormattingOptions(
+        int tabSize = 4
+      , bool insertSpaces = true
```

Comment:
we should document that if `insertSpaces` is `false` it means: use tabs for indentation.

Reply:
Yes, this all needs more documentation still. Although I think @jurgenvinju was not so sure about the meaning of this parameter...

Reply:
We talked about it, and look at the behavior of VS Code, and some discussion based on that.

- `insertSpaces = true` means indent with spaces (of `tabSize` width). Replace tabs used for indentation with spaces.
- `insertSpaces = false` means indent using tabs. Replace spaces used for indentation with tabs.

The realisation was that if you didn't have this behavior, you could never normalize the indentation patterns for a file with mixed tabs&spaces.